egs/ifnenit/v1/local/add_sclite.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+# Apache 2.0
+
+
+#[ -f ./path.sh ] && . ./path.sh
+#. parse_options.sh || exit 1;
+
+if [ $# -ne 3 ] && [ $# -ne 4 ]; then
+  echo "Usage: local/add_sclite.sh <score-dir> <words.txt> <ref-trn> [<ref-clevel-trn>]"
+  echo "  sclite evaluation on word and phoneme level for all *.tra files in score-dir"
+  exit 1;
+fi
+
+oldLC=$LC_ALL
+export LC_ALL="en_US.utf8"
```

Comment:
please clarify why you need to do this (if not, what breaks?)
I prefer if possible to have scripts not be dependent on any locale other than LC_ALL, but if it's an external script that's OK.

And please set LC_ALL to C in path.sh.  That is expected by the rest of Kaldi.

Reply:
This script was redundant and I've removed it, but because I am working with unicode format of transcripts I should set LC_ALL to some string that contain "utf8". If I don't do this I get some error like this: `UnicodeDecodeError: 'ascii' codec can't decode byte 0xd8 in position 89: ordinal not in range(128)`

Reply:
If that is an error coming from some kind of python script, there are ways from the python script itself to resolve this.  That's better than relying on external environment variables.
Remember to use the python3 mechanisms for this (for new scripts, we prefer to explicitly use python3, via #!/usr/bin/env python3).

Reply:
I've tried every solution that I've found but I still get same error, even when I set `PYTHONIOENCODING = utf8`. I think this error might be of terminal encoding then I should set `LC_ALL="en_US.utf8` for those specific lines of script that have a pipeline for processing some UTF-8 text.

Reply:
Can you please provide small example (expected input, expected output and script) that needs LC_ALL to be set? I'll have a look at it -- I don't have time to debug this from the recipe ATM.
The problem with system locale (LC_*) is that it needs locale database for that given combination to be installed and compiled and it's not guaranteed -- so unless there are very specific reasons (and so far it does not look like they are), it's better to fix the python script (which will be more portable).
y.

Reply:
Thank you @jtrmal . I've removed every LC_ALL statements.